### DISCUSSION TITLE
Black to transparent on existing gif, problem with dispose

### DISCUSSION THREAD

**Discussion by philippedubost** [ImageMagick/ImageMagick#6038] [Digital Image Processing]
Hello ! 

>  ImageMagick 7.0.10-62 Q16 x64 2021-02-07  Windows 11

trying to clear the black pixels to transparent pixels on GIFs like this one
<img src="https://user-images.githubusercontent.com/1922509/217197351-2167d384-b65b-4e86-9959-2b9539dcd634.gif" width="100">

I tried 
`magick convert black.gif -transparent black test1.gif`
<img src="https://user-images.githubusercontent.com/1922509/217197895-45efac47-1b98-4d23-bd83-6014063d524c.gif" width="100">

`magick convert -dispose background black.gif -transparent black test2.gif`
<img src="https://user-images.githubusercontent.com/1922509/217197648-c9ddc28e-ecc7-4ccc-a19a-8540e1770764.gif" width="100">
somehow the clearing of the frames is affected by the previous frame

`magick convert -dispose previous black.gif -transparent black test3.gif`
<img src="https://user-images.githubusercontent.com/1922509/217198080-b74fe1ff-a134-40a5-af3d-1134164df1c4.gif" width="100">
same result 

My goal is to get this (done online with [ezgif](https://ezgif.com/effects/)): 
<img src="https://user-images.githubusercontent.com/1922509/217198790-69718439-0216-467c-889f-03af93501d4f.gif" width="100">

thanks a lot for your help !

**Comment by fmw42** [comment]
Remove the convert. In IM 7 use just magick, rather than convert or magick convert.  Second add lots of fuzz to allow more colors to be made transparent. GIFs use binary transparency. So we need to remove the partial black transparency from the antialiasing.

`magick black.gif -fuzz 45% -transparent black test1.gif`

![test1](https://user-images.githubusercontent.com/13458218/217309554-4e73a8a5-b0c4-46ed-a425-ec842e1e290e.gif)
